Gordon360/ApiControllers/MyScheduleController.cs

```diff
@@ -1,161 +1,173 @@
+using Gordon360.Exceptions.CustomExceptions;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Linq;
+
+namespace Gordon360.Controllers.Api
+{
+    [RoutePrefix("api/myschedule")]
+    [CustomExceptionFilter]
+    [Authorize]
+    public class MyScheduleController : ApiController
+    {
+        //declare services we are going to use.
+        private IProfileService _profileService;
+        private IAccountService _accountService;
+        private IRoleCheckingService _roleCheckingService;
+
+        private IMyScheduleService _myScheduleService;
+
+        public MyScheduleController()
+        {
+            var _unitOfWork = new UnitOfWork();
+            _myScheduleService = new MyScheduleService(_unitOfWork);
+            _profileService = new ProfileService(_unitOfWork);
+            _accountService = new AccountService(_unitOfWork);
+            _roleCheckingService = new RoleCheckingService(_unitOfWork);
+        }
+
+        public MyScheduleController(IMyScheduleService myScheduleService)
+        {
+            _myScheduleService = myScheduleService;
+        }
+
+        /// <summary>
+        ///  Gets all myschedule objects for a user
+        /// </summary>
+        /// <returns>A IEnumerable of myschedule objects</returns>
+        [HttpGet]
+        [Route("")]
+        public IHttpActionResult Get()
+        {
+            var authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            var username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
+
+            var id = _accountService.GetAccountByUsername(username).GordonID;
+
+            var result = _myScheduleService.GetAllForID(id);
+            if (result == null)
+            {
+                return NotFound();
+            }
+            return Ok(result);
+        }
+
+        /// <summary>
+        ///  Gets all myschedule objects for a user
+        /// </summary>
+        /// <returns>A IEnumerable of myschedule objects</returns>
+        [HttpGet]
+        [Route("{username}")]
+        public IHttpActionResult Get(string username)
+        {
+            //probably needs privacy stuff like ProfilesController and service
+            var id = _accountService.GetAccountByUsername(username).GordonID;
+        
+            var result = _myScheduleService.GetAllForID(id);
+            if (result == null)
+            {
+                return NotFound();
+            }
+            return Ok(result);
+        }
+                
+
+        /// <summary>Create a new myschedule to be added to database</summary>
+        /// <param name="mySchedule">The myschedule item containing all required and relevant information</param>
+        /// <returns>Created schedule</returns>
+        /// <remarks>Posts a new myschedule to the server to be added into the database</remarks>
+        // POST api/<controller>
+        [HttpPost]
+        [Route("add")]
```

Comment:
We shouldn't need the word add in the route, since it's the most straightforward POST method

Reply:
Change made